pkg/cluster/deploybaseresources.go

```diff
@@ -200,6 +202,15 @@ func (m *manager) deployBaseResourceTemplate(ctx context.Context) error {
 		m.networkInternalLoadBalancer(azureRegion),
 	}
 
+	if m.doc.OpenShiftCluster.UsesWorkloadIdentity() {
```

Comment:
In addition to adding the new WI resources, we need to skip `clusterServicePrincipalRBAC` (see a few lines above this). It caused another nil pointer dereference in my local testing.